src/main/groovy/com/fidesmo/gradle/plugin/OperationTask.groovy

```diff
@@ -37,16 +37,31 @@ class OperationTask extends FidesmoBaseTask {
 
     def executeOperation(UUID operationId) {
 
-        // execute operation by implementing sec-client flow
-        def client = (new OperationClientImpl()).get(
-            operationId,
-            (Transceiver) new SmartcardioTransceiver(),
-            RetrofitSecClient.client
-        )
-        client.transceive().toBlocking().last()
+        /* semantically both options differ. On android after return
+         * the operation has been started, locally the last apdu was
+         * already send.
+         */
+        if (project.hasProperty('fidesmo.adb_reader')) {
+            // call activity (and therefore sec-client) on android phone
+            def android_home = System.env.ANDROID_HOME
+            def cmd = "${android_home}/platform-tools/adb shell " +
+                "am start -W " +
+                "--es descriptionExtra '${description}' " +
+                "--es operationIdExtra '${operationId}' " +
+                "com.fidesmo.sec.android/.ui.OperationActivity_"
+            def proc = cmd.execute()
+            proc.waitFor()
+        } else {
+            // execute operation by implementing sec-client flow
+            def client = (new OperationClientImpl()).get(
+                operationId,
+                (Transceiver) new SmartcardioTransceiver(),
+                RetrofitSecClient.client)
+            client.transceive().toBlocking().last()
+        }
 
         // check operation result by querying the status service
-        int maxRetries = 9
+        int maxRetries = 30
```

Comment:
Oki, I can see why we need this since we are waiting for the result already before the phone starts transmitting to card.